ldai/client.go

```diff
@@ -42,38 +48,47 @@ func New(sdk ServerSDK) (*Client, error) {
 	}, nil
 }
 
+func (c *Client) logConfigWarning(key string, format string, args ...interface{}) {
+	prefix := "AI config '" + key + "': "
+	c.logger.Warnf(prefix+format, args...)
+}
+
+// Config evaluates an AI config named by a given key for the given context.
+//
+// The config's messages will undergo Mustache template interpolation using the provided variables, which may be
+// nil. If the config cannot be evaluated or LaunchDarkly is unreachable, the default value is returned. Note that
+// the messages in the default will not undergo template interpolation.
```

Comment:
> Note that the messages in the default will not undergo template interpolation.

That's not the case for python, and I'm not sure it's the case for JS either. Might be a case where the spec is under-specified. @kinyoklion thoughts?

Reply:
How does it work if the template interpolation fails?

Reply:
We just have not handled this in python and JS yet. We did it this way in .Net.

I am still not sure if it is a great solution, but we do need to choose some solution.

Reply:
> How does it work if the template interpolation fails?
> > If the config cannot be evaluated or LaunchDarkly is unreachable

If the LDClient is offline, we will return the default value from the variation method. In python, that is the default config value as a dictionary. So we get back something in the AI client that looks like the right shape, and so we can still do the templating for ldctx at least.

That seems like something that would be useful for customers.

Reply:
If your template string is invalid in your default value the interpolation can still fail. Which is a hangup itself.

Reply:
Yes, templating related failures are a problem we need to surface as well. I just want to make sure we're handling at least the case where we could do templating for a default value, but aren't.

Reply:
Closing the loop, we should discuss more, but I won't consider this blocking for getting out the alpha.